tests/CryptoBank.WebAPI.Tests.Integration/Features/Deposits/Jobs/BlockchainDepositScannerTests.cs

```diff
@@ -0,0 +1,182 @@
+using CryptoBank.WebAPI.Tests.Integration.Harnesses;
+using CryptoBank.WebAPI.Tests.Integration.Harnesses.Base;
+using CryptoBank.WebAPI.Tests.Integration.Harnesses.Bitcoin;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.EntityFrameworkCore;
+using NBitcoin;
+using NBitcoin.RPC;
+
+namespace CryptoBank.WebAPI.Tests.Integration.Features.Deposits.Jobs;
+
+public class BlockchainDepositScannerTests : IAsyncLifetime
+{
+    private readonly BitcoinHarness<Program> _bitcoin;
+    private readonly DatabaseHarness<Program, CryptoBankDbContext> _database;
+    private readonly WebApplicationFactory<Program> _factory;
+
+    private CancellationToken _cancellationToken;
+    private CancellationTokenSource _cancellationTokenSource;
+    private IClock _clock;
+    private AsyncServiceScope _scope;
+
+    public BlockchainDepositScannerTests()
+    {
+        _bitcoin = new BitcoinHarness<Program>();
+        _database = new DatabaseHarness<Program, CryptoBankDbContext>();
+
+        _factory = new WebApplicationFactory<Program>()
+            .WithHarness(_bitcoin)
+            .WithHarness(_database)
+            .WithWebHostBuilder(
+                builder =>
+                {
+                    builder.ConfigureAppConfiguration(
+                        (_, configBuilder) =>
+                        {
+                            configBuilder.AddInMemoryCollection(
+                                new Dictionary<string, string>
+                                {
+                                    {
+                                        "Features:Deposits:BitcoinBlockchainScanInterval",
+                                        TimeSpan.FromSeconds(1).ToString()
+                                    },
+                                });
+                        });
+                });
+    }
+
+    public async Task InitializeAsync()
+    {
+        var delay = Debugger.IsAttached ? TimeSpan.FromMinutes(10) : TimeSpan.FromSeconds(60);
+        _cancellationTokenSource = new CancellationTokenSource(delay);
+        _cancellationToken = _cancellationTokenSource.Token;
+
+        await _bitcoin.Start(_factory, _cancellationToken);
+        await _database.Start(_factory, _cancellationToken);
+
+        var _ = _factory.Server;
+        _scope = _factory.Services.CreateAsyncScope();
+
+        _clock = _scope.ServiceProvider.GetRequiredService<IClock>();
+    }
+
+    public async Task DisposeAsync()
+    {
+        await _scope.DisposeAsync();
+        await _factory.DisposeAsync();
+
+        await _database.Stop();
+        await _bitcoin.Stop();
+    }
+
+    private static async Task Mine50Btc(RPCClient client)
+    {
+        var bitcoinAddress = await client.GetNewAddressAsync(new GetNewAddressRequest());
+
+        await client.GenerateToAddressAsync(101, bitcoinAddress);
+    }
+
+    private async Task WaitForCryptoDepositCreation(int expectedCount)
+    {
+        while (true)
```

Comment:
No, it doesn't hang forever because I use `CancellationToken` in the method `CryptoDeposits.CountAsync(_cancellationToken)`. I met that situation during development, It worked well.